### PR TITLE
Convert rst to markdown - profiler.rst #155031

### LINKED ISSUE #155031: Convert to markdown: onnx_verification.rst, onnx.rst, optim.rst, package.rst, profiler.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [onnx_verification.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx_verification.rst)
- [onnx.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/onnx.rst)
- [optim.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/optim.rst)
- [package.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/package.rst)
- [profiler.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/profiler.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by GdoongMathew**
Fixes https://github.com/pytorch/pytorch/issues/155031

* [profiler.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/profiler.rst)

cc @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by AlannaBurke** [comment]
Preview links: 

https://docs-preview.pytorch.org/pytorch/pytorch/155559/optim.html
https://docs-preview.pytorch.org/pytorch/pytorch/155559/profiler.html

File: docs/source/optim.md [hunk 1]
```diff
@@ -0,0 +1,707 @@
...
+on a given dataloader `loader` at the end of training:
+
+```python
+>>> torch.optim.swa_utils.update_bn(loader, swa_model)
+```
+
+{func}`update_bn` applies the `swa_model` to every element in the dataloader and computes the activation
+statistics for each batch normalization layer in the model.
+
+```{warning}
```
Comment (AlannaBurke) [inline docs/source/optim.md]: This warning is showing like a code block, I don't think it needs to be indented. 
Monosnap torch optim — PyTorch main documentation 2025-06-10 13-58-18

**Comment by svekars** [comment]
three more files are coming, right?

**Review by AlannaBurke — CHANGES_REQUESTED** [review]
One comment on optim.md.
profiler.md looks good!

**Comment by AlannaBurke** [comment]
@svekars they're in another PR, I'm reviewing it.

File: docs/source/optim.md (hunk 1 above)
Comment (GdoongMathew) [reply docs/source/optim.md]: Thanks!! That was a mistake, which should be fixed in https://github.com/pytorch/pytorch/pull/155559/commits/0e2392ac0c049bfe755256c262b7c35a81bc6ffa

The following are the results from the local build.

image

image

**Comment by svekars** [comment]
https://docs-preview.pytorch.org/pytorch/pytorch/155559/optim.html has changed the Created On date - need to make sure this is fixed. Maybe use VSCode git rename?

**Review by AlannaBurke — CHANGES_REQUESTED** [review]
Looks good but need to fix the created date on optim.md

**Comment by svekars** [comment]
Here is how to fix:
1) Save your changes to optim.md somewhere.
2) On this branch undo your changes optim - delete it and checkout optim.rst. From docs dir: git checkout origin/main -- source/optim.rst
3) git mv source/optim.rst source/optim.md
4) Make a commit: git commit -m "Renamed source/optim.rst to source/optim.md". 
5) Can check git log: `git log -- follow -- optim.md` - this should show you a bunch of commits. 
6) Push into this PR the above commit.
7) Copy paste changes you've saved in step 1 into optim.md. Commit/push into this PR.

**Comment by GdoongMathew** [comment]
Not sure why this happened, 

image

But I still got the same date even after following the instruction. Probably I'll need to remove this file in this PR and open a new one to check if the problem still persists.

**Comment by GdoongMathew** [comment]
I just opened #155813, which should fix the created date problem here. Will remove any change relate to `optim.rst` in this PR. Sorry for all the troublesome.

**Review by svekars — APPROVED** [review]
LGTM
